src/target/riscv/riscv.c

```diff
@@ -3675,6 +3700,16 @@ static const struct command_registration riscv_exec_command_handlers[] = {
 			"expose. custom0 is accessed as abstract register number 0xc000, "
 			"etc. This must be executed before `init`."
 	},
+	{
+		.name = "hide_csrs",
+		.handler = riscv_set_hidden_csrs,
+		.mode = COMMAND_CONFIG,
+		.usage = "n0[-m0|=name0][,n1[-m1|=name1]]...",
```

Comment:
I recommend the following syntax for the list of hidden CSRs: `{n0|n0-m0|name0}[,n1|n1-m1|name1]...`

Or even a simpler variant, without the name, would suffice: `{n0|n-m0}[,n1|n-m1]...`

The current syntax (implemented in `parse_ranges()`) makes sense for `expose_csrs` command but not for `hide_csrs`. In `hide_csrs`, there is no point in setting a custom name for a certain CSR number (`some_num=some_name`).

Reply:
addressed (I should mention that I haven't touched the actual parsing function, though - hopefully it's not a big deal)